x/ecocredit/genesis.go

```diff
@@ -1,172 +0,0 @@
-package ecocredit
-
-import (
-	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
-	"github.com/regen-network/regen-ledger/types/math"
-)
-
-// Validate performs basic validation for each credit-batch,
-// it returns an error if credit-batch tradable or retired supply
-// does not match the sum of all tradable or retired balances
```

Comment:
Do we need batch balance = batch supply validation? To validate the genesis state we need a proto message for each type.

```
message Classes {
  repeated ClassInfo classes = 1;
}
```

Reply:
Hmmm... I'm not sure how to best handle this. cc @aaronc @technicallyty @robert-zaremba

Reply:
going off what aaron mentioned below, i think we should be able to put the raw json inside an in-mem ORM db, then use iterate over the batch balance/supply tables and check invariants that way